### PR TITLE
Added Verify RPC call for OS client

### LINKED ISSUE #125: Implement OS Verify RPC for Client

### PR DESCRIPTION
**Pull request by ericm**
Related Issue: Closes #125

### CODE REVIEW

**Comment by ericm** [comment]
I decided to return activation fail separate to error since it indicates a previous error rather than a current one

File: gnoi/os/client.go [hunk 1]
```diff
@@ -55,3 +55,16 @@ func (c *Client) Activate(ctx context.Context, version string) error {
 	}
 	return nil
 }
+
+// Verify invokes the Verify RPC for the OS service and it returns the running version and activation fail message or an error.
+func (c *Client) Verify(ctx context.Context) (string, string, error) {
```
Comment (samribeiro) [inline gnoi/os/client.go:60]: One of the few cases it is worth naming the return variables.

```
func (c *Client) Verify(ctx context.Context) (version, failMsg string, err error) {
        var out *pb.VerifyResponse
	if out, err = c.client.Verify(ctx, &pb.VerifyRequest{}); err != nil {
		return
	}
	version = out.GetVersion()
        failMsg = out.GetActivationFailMessage()
	return
}
```

Comment (ericm) [reply gnoi/os/client.go:60]: Oh that's a great idea

**Comment by ericm** [comment]
Should be done :)

File: gnoi/os/client_test.go [hunk 2]
```diff
@@ -36,10 +38,20 @@ type activateTest struct {
...
+	failMessage string
+}
+
 func (c *mockClient) Activate(ctx context.Context, in *pb.ActivateRequest, opts ...grpc.CallOption) (*pb.ActivateResponse, error) {
 	return c.activate(ctx, in, opts...)
 }
 
+func (c *mockClient) Verify(ctx context.Context, in *pb.VerifyRequest, opts ...grpc.CallOption) (*pb.VerifyResponse, error) {
+	return c.verify(ctx, in, opts...)
+}
```
Comment (loriscn12) [inline gnoi/os/client_test.go:53]: I like the approach ;)

File: gnoi/os/client_test.go [hunk 3]
```diff
@@ -36,10 +38,20 @@ type activateTest struct {
 	wantErr bool
 }
 
+type verifyTest struct {
+	name,
+	runningVersion,
+	failMessage string
+}
```
Comment (samribeiro) [inline gnoi/os/client_test.go:45]: you are only using this once, perhaps making it inline downthere is better and more readable?

```
tests := []struct {
        name,
        runningVersion,
        failMessage string
}{
        {"Is Running", "version", ""},
        {"Previous Activation Fail", "version", "Activation fail"},
}
```

**Comment by ericm** [comment]
Done :)

**Pull request merged**